hazelcast-client/src/main/java/com/hazelcast/client/impl/clientside/ClientImpl.java

```diff
@@ -48,4 +54,19 @@ public InetSocketAddress getSocketAddress() {
     public ClientType getClientType() {
         return ClientType.JAVA;
     }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public String getAttribute(String attributeKey) {
+        return attributes.get(attributeKey);
+    }
+
+    @Override
```

Comment:
Is this map exposed to the public directly? In other words: could it be that it gets mutated?

Reply:
I made it unmodifiable in the constructor.